core/codegen/src/syn_ext.rs

```diff
@@ -81,17 +81,15 @@ impl IdentExt for syn::Ident {
     }
 
     fn uniqueify_with<F: FnMut(&mut dyn Hasher)>(&self, mut f: F) -> syn::Ident {
-        use std::sync::atomic::{AtomicUsize, Ordering};
-        use std::collections::hash_map::DefaultHasher;
-
-        // Keep a global counter (+ thread ID later) to generate unique ids.
-        static COUNTER: AtomicUsize = AtomicUsize::new(0);
-
-        let mut hasher = DefaultHasher::new();
-        self.hash(&mut hasher);
-        std::process::id().hash(&mut hasher);
-        std::thread::current().id().hash(&mut hasher);
-        COUNTER.fetch_add(1, Ordering::AcqRel).hash(&mut hasher);
+        // Use rustc_hash because it's deterministic, whereas the DefaultHasher
+        // in std is nondeterministic. This prevents problems where compiling
```

Comment:
I don't believe this is necessarily true. [`DefaultHasher::new()`](https://doc.rust-lang.org/stable/std/collections/hash_map/struct.DefaultHasher.html) always returns a `SipHasher` with keys of `0`, so it should always hash to the same values and should thus be deterministic.